flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/testutil/HBaseTestCluster.java

```diff
@@ -0,0 +1,328 @@
+            CompletableFuture.runAsync(cluster::waitUntilShutDown).get(240, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("Interrupted while waiting for HBase test cluster to shut down", e);
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+            LOG.error("Exception while waiting for HBase test cluster to shut down", e);
+        } catch (TimeoutException e) {
+            e.printStackTrace();
+            LOG.error("Waiting for HBase test cluster to shut down timed out", e);
+        }
+    }
+
+    public boolean canConnectToCluster() throws InterruptedException, ExecutionException {
+        try {
+            return CompletableFuture.supplyAsync(
+                            () -> {
+                                try (Connection connection =
+                                        ConnectionFactory.createConnection(getConfig())) {
+                                    return true;
+                                } catch (IOException e) {
+                                    LOG.error("Error trying to connect to cluster", e);
+                                    return false;
+                                }
+                            })
+                    .get(10, TimeUnit.SECONDS);
+        } catch (TimeoutException e) {
+            LOG.error("Trying to connect to HBase test cluster timed out", e);
+            return false;
+        }
+    }
+
+    public void clearTables() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            for (TableDescriptor table : admin.listTableDescriptors()) {
+                admin.disableTable(table.getTableName());
+                admin.deleteTable(table.getTableName());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Could not clear test cluster tables", e);
+        }
+    }
+
+    public void clearReplicationPeers() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            StringBuilder logMessage = new StringBuilder("Clearing existing replication peers:");
+            for (ReplicationPeerDescription desc : admin.listReplicationPeers()) {
+                logMessage.append("\n\t").append(desc.getPeerId()).append(" | ").append(desc);
+                admin.removeReplicationPeer(desc.getPeerId());
+            }
+            LOG.info(logMessage.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Could not clear test cluster replication peers", e);
+        }
+    }
+
+    public List<ReplicationPeerDescription> getReplicationPeers() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            return admin.listReplicationPeers();
+        } catch (IOException e) {
+            LOG.error("Error retrieving replication peers", e);
+            return null;
```

Comment:
Afaik no code path can handle null.

It also sounds as if a test failure would be more appropriate for example by throwing `new AssertionError(..., e)`.

Reply:
✅ ❓  in 902061f240103e2c5680a27f2aa1d2bef9c5ac58 - using `RuntimeException` for code readability, as this method is called in some runnables. Potentially, `ThrowingRunnable.unchecked` could be an alternative solution?

Reply:
✅ in 98b0735381041659c652c17f648e099a92d2746b - now using `ThrowingRunnable.unchecked` and properly throwing IOException